### PR TITLE
:arrow_up: Pre-0.0.2 housekeeping

### PR DESCRIPTION
**Pull request by JMicheli**
This pull request is meant to do some general project housekeeping tasks.

It:
* Bumps various crate and node package versions that could be increased with only minor refactoring;
* Addresses a todo by refactoring the list_directory function;
* Addresses a todo by adding removed thumbnail counting;
* Addresses a todo by providing a description for `ThumbnailGenerationJob`; and
* Addresses a todo by making `PauseJob` and `ResumeJob` take an `AcknowledgeableCommand` parameter.

### CODE REVIEW

File: apps/server/src/routers/api/v1/filesystem.rs [hunk 1]
```diff
@@ -48,23 +48,21 @@ pub(crate) fn mount(app_state: AppState) -> Router<AppState> {
...
-		#[cfg(target_os = "windows")]
-		return "C:\\".to_string();
-		#[cfg(target_family = "unix")]
-		return "/".to_string();
-	});
-	let start_path = Path::new(&start_path);
+	let start_path = match input.path {
+		Some(path) => PathBuf::from(path),
+		None => get_os_start_path(),
+	};
```
Comment (aaronleopold) [inline apps/server/src/routers/api/v1/filesystem.rs:65]:
```suggestion
	let start_path = input
		.path
		.map(PathBuf::from)
		.unwrap_or_else(get_os_start_path);
```

File: apps/server/src/routers/api/v1/filesystem.rs [hunk 2]
```diff
@@ -133,3 +103,36 @@ pub async fn list_directory(
...
+	let path = entry.path();
+	let stem = path.file_stem().unwrap_or_default();
+
+	// Remove hidden files starting with period
+	if stem.to_str().unwrap_or_default().starts_with('.') {
+		return None;
+	}
+
+	Some(entry)
+}
```
Comment (aaronleopold) [inline apps/server/src/routers/api/v1/filesystem.rs:138]: Not important for this PR, but might be worth exporting `PathUtils` trait from `stump_core` so we don't maintain duplicate logic, e.g. `PathUtils::is_hidden_file`.

File: core/src/filesystem/directory_listing.rs [hunk 3]
```diff
@@ -25,23 +29,31 @@ pub struct DirectoryListing {
...
+		}
+	}
+
+	pub fn from_pathbuf(path: PathBuf) -> Self {
+		Self {
+			name: path.file_parts().file_name,
+			is_directory: path.is_dir(),
+			path,
 		}
 	}
```
Comment (aaronleopold) [inline core/src/filesystem/directory_listing.rs:50]: I think this is a good candidate for a `From` trait, e.g. `Impl From<PathBuf> for DirectoryListingFile`

**Review by aaronleopold — COMMENTED** [review]
This was a huge effort, I know managing dependency upgrades can be a bit tedious so I really appreciate you doing this!

I had a couple of nit picks of little importance, the once thing keeping me from approving and merging is that I think we need to bump the minimum rust version with these changes. Using the repo's configured `1.72.1` toolchain locally reported the following:

- jsonwebtoken requires 1.73.0
- clap_builder requires 1.74.0

**Comment by aaronleopold** [comment]
> Any reason not to bump it up to 1.77.2 (the current stable release)?

I don't think so, might as well give it a shot and see what CI says

**Comment by aaronleopold** [comment]
@JMicheli I updated the toolchain and a few other miscellaneous things (e.g. pdfium version, lints after updates, merge conflicts, etc). 

There was actually a clippy bug I found, too! 

Let me know if there's anything you want to add to this PR. I'll plan to merge this tomorrow after work, get a nightly image out, then 0.0.2 over the weekend. Thanks again for doing all of this cleanup!

**Comment by JMicheli** [comment]
I won't have time today to do anything further, I say merge it if you're good to do so.

**Pull request merged**